client/src/accounts/authentication/register.hook.test.tsx

```diff
@@ -35,7 +39,7 @@ describe('useRegisterHook', () => {
     act(() => result.current.mutate(registerForm));
 
     await waitFor(() => expect(result.current.isError).toBe(true));
-    expect(result.current.error).toEqual({ message: expectedMessage });
+    expect(result.current.error).toStrictEqual({ message: expectedMessage });
```

Comment:
What made you change to this? I had hoped it was `eslint-plugin-jest-dom`, but having undone this change, nothing identified this as to be changed. The point of the jest-dom plugin is to coax us to use the right matcher, having looked into it further, there is another plugin `eslint-plugin-jest`, but we don't use that one. The reason why we use the jest-dom one is because it is referenced in the [testing-library docs](https://testing-library.com/docs/ecosystem-eslint-plugin-jest-dom). I'm not saying this is wrong, just trying to understand how to got to the decision to make this change.

Reply:
sorry, my bad. I'd misread your other comment and had replaced the toEqual() with toBe() for the object. This of course failed - it's not a basic type - but it suggested using toStrictEqual() instead. The tests passed.

Looking at the docs, toStrictEqual() does a deep-comparison like toEqual(), but also checks the type. As I'm not specifying a type for the error message, the two will be identical. I should either (a) keep toStrictEqual but add a type for the return message or (b) revert to toEqual()

Reply:
I would like us to get guidance from eslint tools, there are plugins that statically analyse the code and can suggest a better matcher e.g. you use `toEqual` on a basic type and it shows a warning/error stating to change this to `toBe` or vice versa,that kind of thing. If it is the IDE intellisense, then I suspect we are all using different extensions and will get different results, I'd like the advice to be consistent if we can.

Reply:
We have this capability in orbis, I'm just not convinced it is working in our `app-auth` client, even though it is in the eslint rules. I need to do a bit of digging into this, to prove it either way.